instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -245,12 +245,25 @@ def response_hook(span: Span, status: str, response_headers: List):
 from typing import Collection
 
 import flask
+import opentelemetry.instrumentation.wsgi as otel_wsgi
 from packaging import version as package_version
 
-import opentelemetry.instrumentation.wsgi as otel_wsgi
 from opentelemetry import context, trace
 from opentelemetry.instrumentation.flask.package import _instruments
 from opentelemetry.instrumentation.flask.version import __version__
+
+try:
+    flask_version = flask.__version__
+except AttributeError:
+    try:
+        from importlib import metadata
+    except ImportError:
+        import importlib_metadata as metadata
+    try:
+        flask_version = metadata.version("flask")
+    except (ImportError, metadata.PackageNotFoundError):
+        flask_version = "unknown"
```

Comment:
I think `unknown` may fail to parse below on L287 and then give a confusing exception. It would be better to just raise a specific exception message here about not being able to detect flask version, or just not even surround with `try` here and let the exception bubble out

Reply:
thank you for your feedback, so i tried to include your feedback.